connection.js

```diff
@@ -269,45 +269,23 @@ class Connection {
         .catch(conn_log);
 
 
-    }
-    // Check challenge entirely, and return reject status + optional error msg.
-    //
-    checkChallenge(notification) {
-
-        let result = {};
-        for (let test of [this.checkChallengeMandatory,
-                          this.checkChallengeSanityChecks,
-                          this.checkChallengeBooleans,
-                          this.checkChallengeAllowedFamilies,
-                          this.checkChallengeSettings]) {
-            result = test.bind(this)(notification);
-            if (result.reject) return result;
-        }
-
-        return { reject: false };  /* All good. */
-
     }
     // Check challenge mandatory conditions
     //
-    checkChallengeMandatory(notification) {
+    checkChallengeMandatory(notification, config_r_u, r_u_strings) {
 
         // check user is acceptable first, else don't mislead user (is professional is in booleans below, not here):
-        for (let uid of ["username", "id"]) {
-            if (config.banned_users[notification.user[uid]]) {
-                return bannedFamilyReject("bans", uid, notification.user[uid]);
-            }
-            if (notification.ranked && config.banned_users_ranked[notification.user[uid]]) {
-                return bannedFamilyReject("bansranked", uid, [notification.user[uid]]);
-            }
-            if (!notification.ranked && config.banned_users_unranked[notification.user[uid]]) {
-                return bannedFamilyReject("bansunranked", uid, [notification.user[uid]]);
-            }
+        if (checkUserIsInSet(notification.user, config_r_u["blacklist"])) {
+            return getBlackWhiteListReject("blacklist", r_u_strings.for_r_u_games);
         }
-        const resultRank = minMaxHandicapRankRejectResult("rank", notification.user.ranking, false, notification.ranked);
-        if (resultRank) return resultRank;
+        if (!checkUserIsInSet(notification.user, config_r_u["whitelist"])) {
```

Comment:
Here we should also check if the set contains items, if it contains no items then bot admin is not using the whitelist.

Reply:
oh actually if whitelist is empty then the check returns false, so !check is true, so we'd always reject !!
i wanted to avoid a repetition but changed the meaning of the code !
thanks, fixed !